django/db/models/expressions.py

```diff
@@ -1340,6 +1340,10 @@ def __len__(self):
     def __iter__(self):
         return iter(self.get_cols())
 
+    def __repr__(self):
+        cols = ", ".join(repr(col) for col in self.get_cols())
+        return "{}({})".format(self.__class__.__name__, cols)
```

Comment:
I think we should use an f string:

```suggestion
        return f"{self.__class__.__name__}({cols})
```

Or if we don't like the attribute lookup within the string:

```suggestion
        name = self.__class__.__name__
        return f"{name}({cols})
```

Finally, if we're confident `ColPairs` won't be subclassed:

```suggestion
        return f"ColPairs({cols})
```

Reply:
I'm not sure about this @LilyFoote, I was following `Col.__repr__`, but it seems like the `__repr__` functions in `expressions.py` are completely inconsistent, some use `f-strings`, some use `%s` and some use `.format()`. 😅

Reply:
`__repr__` [should represent a valid instance](https://docs.python.org/3/reference/datamodel.html#object.__repr__) (if possible), so `ColPairs({cols})` would be misleading. Check out all commit related with ticket-24121. You have everything in place, so:
```
f"{self.__class__.__name__}({self.alias}, {self.targets}, {self.sources}, {self.output_field})"
```
would be the correct representation.

Reply:
> I'm not sure about this @LilyFoote, I was following `Col.__repr__`, but it seems like the `__repr__` functions in `expressions.py` are completely inconsistent, some use `f-strings`, some use `%s` and some use `.format()`. 😅

I think it's inconsistent because avoiding churn is more important, but I think f-strings are preferred for new code.

Reply:
I adjusted but if we're aiming for a valid instance, shouldn't the arguments also use `repr` (or `!r`) instead of `str`? I see `str` is used in most `__repr__` functions.

Reply:
Yes, I'd use `{something!r}` here.